File: 45.jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int max_pos = 0;
        int min_pos = 0;
        int step = 0;
        while (max_pos < nums.size() - 1) {
            step++;
            int next_max_pos = max_pos + 1;
            for (int i = min_pos; i <= max_pos; i++) {
                next_max_pos = max(next_max_pos, i + nums[i]);
            }
            min_pos = max_pos + 1;
            max_pos = next_max_pos;
        }
        return step;
    }
};

File: 45.jump-game-ii.py
class Solution:
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_pos = 0
        min_pos = 0
        step = 0
        while max_pos < len(nums) - 1:
            step += 1
            next_max_pos = max_pos + 1
            for i in range(min_pos, max_pos + 1):
                next_max_pos = max(next_max_pos, i + nums[i])
            min_pos = max_pos + 1
            max_pos = next_max_pos
        return step